Formula/n/netaddr.rb

```diff
@@ -0,0 +1,26 @@
+class Netaddr < Formula
+  include Language::Python::Virtualenv
+
+  desc "Network address manipulation library"
+  homepage "https://netaddr.readthedocs.io/en/latest/"
+  url "https://files.pythonhosted.org/packages/54/e6/0308695af3bd001c7ce503b3a8628a001841fe1def19374c06d4bce9089b/netaddr-1.2.1.tar.gz"
+  sha256 "6eb8fedf0412c6d294d06885c110de945cf4d22d2b510d0404f4e06950857987"
+  license "BSD-3-Clause"
+  head "https://github.com/netaddr/netaddr.git", branch: "master"
+
+  depends_on "python@3.12"
+
+  resource "setuptools" do
```

Comment:
is this actually needed? i don't see it importing any of the packages it provides: https://github.com/search?q=repo%3Anetaddr%2Fnetaddr+%2Fdistutils%7Csetuptools%7Cpkg_resources%2F&type=code

Reply:
Originally I just had it building with `*std_pip_args`, but it failed without `setuptools`. Hence the switch to this.

Reply:
Is this related to what you mentioned in this comment? https://github.com/Homebrew/homebrew-core/pull/163089#discussion_r1493457732

Reply:
yup same thing here

Reply:
A while back all the `virtualenv` formulae that had no resources were converted over to `*std_pip_args`. I wonder what's the best way to handle this now.

Reply:
Yea I think we converted too many to use `std_pip_args`. For a python application, we should always use a venv (as our [docs explain](https://docs.brew.sh/Python-for-Formula-Authors#installing-applications)):

* It doesn't pollute the user's global packages (`std_pip_args` installs by default to `lib` and whereas venv installs to `libexec` and only exposes the binaries. this is consistent with how we handle other languages, for example `node`-based formulae)
* It's more maintainable (if later releases add dependencies, no changes required. otherwise, we risk shipping broken formulae)
* The only overhead is a few python symlinks

But for a self-contained python library / bindings, then `std_pip_args` makes sense

Reply:
Makes sense to me. Thanks for the explanation. I've pulled the `setuptools` resource out.